core/src/memory/details/unique_alloc_deleter_impl.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <cassert>
+
+#include "vecmem/memory/unique_ptr.hpp"
+
+namespace vecmem::details {
+unique_alloc_deleter_impl::unique_alloc_deleter_impl(void) = default;
```

Comment:
Huhh... I didn't know that you could do this... But yeah, let's put the default implementations into the `.cpp` file!

I would just make the other constructors/assignment functions "default" as well.